src/authentication/UnsecureWebIdExtractor.ts

```
@@ -6,7 +7,18 @@ import { CredentialsExtractor } from './CredentialsExtractor';
  * Credentials extractor which simply interprets the contents of the Authorization header as a webID.
  */
 export class UnsecureWebIdExtractor extends CredentialsExtractor {
-  public async handle(input: HttpRequest): Promise<Credentials> {
-    return { webID: input.headers.authorization };
+  protected readonly logger = getLoggerFor(this);
+
+  public async canHandle({ headers }: HttpRequest): Promise<void> {
+    const { authorization } = headers;
+    if (!authorization || !authorization.startsWith('WebID ')) {
+      throw new Error('No WebID Authorization header specified.');
```

This should throw an `UnsupportedHttpError` (or a more specific one).